health/state.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package health
+
+import (
+	"time"
+)
+
+// State is a representation of the health status of the backend, which is
+// provided to the client UI via LocalAPI. It is used to simplify the data structure
+// provided over JSON.
+type State struct {
+	// Warnings maps each WarnableCode to a WarningStateRepresentation.
+	Warnings map[WarnableCode]WarningStateRepresentation
+}
+
+type WarningStateRepresentation struct {
```

Comment:
all exported (capitalized) things need docs

But also this is a mouthful of a name

This can't just be WarningState? Oh! this is the Warning+WarningState combo one. (document that part)

The better name for this one would be just "WarningState".

The one you currently call "WarningState" in this PR could really be unexported. And it'd clean up the call sites where you mark things unhealthy. You don't need the callers to set the BrokenSince time anyway (that's private to the health package) so you really just need the args. And then maybe you make a type like:

package health
type Args map[string]string

So you can write:

     ...SetUnhealthy(health.Args{...}) or SetUnhealthy(nil) for no args